File: typescript/Behavioral/16-Mediator/Colleague/Scheduler.ts
import { Colleague } from "./Colleague";
import { IDispatchMediator } from "../Mediator/IDispatchMediator";

export class Scheduler extends Colleague {
  private slotCounter = 1;

  constructor(mediator: IDispatchMediator) {
    super(mediator, "Scheduler");
  }

  bookSlot(category: string): string {
    const slot = `Slot-${this.slotCounter++} (${category})`;
    console.log(`  [Scheduler] Booked: ${slot}`);
    return slot;
  }
}

File: csharp/Behavioral/16-Mediator/Colleague/Scheduler.cs
public class Scheduler : Colleague
{
    private int _slotCounter = 1;

    public Scheduler(IDispatchMediator mediator)
        : base(mediator, "Scheduler") { }

    public string BookSlot(string category)
    {
        var slot = $"Slot-{_slotCounter++} ({category})";
        Console.WriteLine($"  [Scheduler] Booked: {slot}");
        return slot;
    }
}
